tests/test_proxy.py

```diff
@@ -167,3 +171,127 @@ def test_proxy_500_misconfiguration(self):
         self.assertEqual(p.res.headers['Content-Type'], 'application/json')
         self.assertIn('Proxy error while generating URL to request',
                       p.res.body)
+
+
+class TestServerErrorHandling(unittest.TestCase):
+    def setUp(self):
+        self.conf = config.Conf()
+        self.conf.host = "localhost"
+        self.conf.scheme = "http"
+        self.conf.port = 8000
+
+    def make_request(self, method="GET", path_query="/", headers=None):
+        req = proxy.Req()
+        req.method = method if method else "GET"
+        req.path_query = path_query if path_query else "/"
+        req.headers = headers if headers else {"Accept": "application/json"}
+        return req
+
+    def test_cannot_connect(self):
+        """
+        Test for "502 Bad Gateway" when the server can't be reached.
```

Comment:
great tests! note to other reviewers that this provides regression coverage for #13